screens/dashboard/components/goal.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import commonStyles from '../../../styles/common';
+import PropTypes from 'prop-types';
+import {
+    StyleSheet,
+    Text,
+    View  } from 'react-native';
+const styles = StyleSheet.create(commonStyles);
+
+export class Goal extends React.Component {
+
+    static propTypes = {
+        title: PropTypes.string.isRequired,
+        detail: PropTypes.string.isRequired,
+    }
+
+    //TODO: needs safer key .. perhaps add id value to goal object
```

Comment:
Like I said, our data model is not fully baked (maybe even not half-baked), but we can totally get an id associated with these goals.

I believe there are 2 goal arrays; one for upcoming & one for completed goals. I'd have to look at the code to tell which one is which, but one of the two types already has an id field.  However, it might be better to just use the array element index since that's auto-generated (I think we'd have to generate the id field ourselves).

As I mentioned, you can go in and modify the user data as necessary (just be careful since it's our only copy during development):

![image](https://user-images.githubusercontent.com/1809882/60843921-f72b5a00-a1a5-11e9-89a3-bb5aaa6a490c.png)

Reply:
I think I actually added those id fields in firebase while working on a follow up commit.  My understanding is that React uses the "key" value to determine if anything has changed during its render cycle, so using the array index might cause weird behavior if we are manipulating the data (I can test it though :) )

Reply:
Thanks .. have been out of town, but will follow up with more changes that add the status stuff in the next few days!  Will submit the pull request then

Reply:
FYI
 I'm fully baking the Goal model.  for issue #93.  I'll be moving Goals from the array to a sub-collection.  This is needed so @iritush can add the right listeners.  

Also, when I'm done, you wont need the id field.  Firebase will assign an id when the goal is added.

None of this should affect @mpettit 's work.